addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,199 @@
+                line.amount = 0.0
+            else:
+                line.amount, dummy, dummy = line._tax_compute_all_helper()
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    withholding_ids = fields.One2many(
+        'account.payment.register.withholding', 'payment_register_id', string="Withholdings",
+        compute='_compute_withholdings', readonly=False, store=True)
+    net_amount = fields.Monetary(compute='_compute_net_amount', readonly=True,  help="Net amount after withholdings")
+
+    @api.depends('line_ids', 'can_group_payments', 'group_payment')
+    def _compute_withholdings(self):
+        supplier_recs = self.filtered(lambda x: x.partner_type == 'supplier' and (not x.can_group_payments or (x.can_group_payments and x.group_payment)))
+        for rec in supplier_recs:
+            taxes = rec._get_withholding_tax()
+            rec.withholding_ids = [Command.clear()] + [Command.create({'tax_id': x.id, 'base_amount': 0}) for x in taxes]
+        (self - supplier_recs).withholding_ids = False
+
+    @api.depends('withholding_ids.amount', 'amount', 'l10n_latam_check_id')
+    def _compute_net_amount(self):
+        for rec in self:
+            if rec.l10n_latam_check_id:
+                rec.currency_id = rec.l10n_latam_check_id.currency_id
+                rec.net_amount = rec.l10n_latam_check_id.amount
+                net_info = rec._get_amount_net_info()
+                witholding_net_amount = min(rec.l10n_latam_check_id.amount , net_info[1])
+                rec.amount = witholding_net_amount * net_info[0] + (rec.l10n_latam_check_id.amount - witholding_net_amount)
+            else:
+                rec.net_amount = rec.amount - sum(rec.withholding_ids.mapped('amount'))
+
+    def _get_withholding_tax(self):
+        self.ensure_one()
+        return self.line_ids.move_id.invoice_line_ids.product_id.l10n_ar_supplier_withholding_taxes_ids.filtered(
+                lambda y: y.company_id == self.company_id)
+
+    def _get_amount_net_info(self):
+        self.ensure_one()
+        # This method allows to simulate the total net amount of a payment and the relationship between the total and net amounts.
+        # return a tupple
+        #   - factor between total and net
+        #   - Net amount total
+        wizard = self.env['account.payment.register'].with_context(active_model='account.move.line', active_ids=self.line_ids.ids).new()
+        wizard.currency_id = self.l10n_latam_check_id.currency_id
+        wizard.payment_date = self.payment_date
+        wizard.withholding_ids = [Command.clear()] + [Command.create({'tax_id': x.tax_id.id, 'base_amount': x.base_amount , 'amount': x.amount}) for x in self.withholding_ids]
+        return (wizard.amount / wizard.net_amount, wizard.net_amount)
+
+    def _get_withholding_move_line_default_values(self):
+        return {
+            'partner_id': self.partner_id.id,
+            'currency_id': self.currency_id.id,
```

Comment:
a bit weird as both methods should be set from the move already as they are compute.

Reply:
check if really needed

Reply:
we've checked and we can remove the partner but the currency is still needed

Reply:
it seems it is setting the company currency instead of the move's indeed

Reply:
But still not need for this separate method then...